HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -528,26 +525,17 @@ public IActionResult Pending_Get(QuestionFlowStep? entryPoint, bool emailPrefere
     }
 
     [HttpPost("pending")]
-    public async Task<IActionResult> Pending_Post(IneligibleViewModel viewModel)
+    public async Task<IActionResult> Pending_Post(PendingViewModel viewModel)
     {
         if (!ModelState.IsValid)
         {
-            return Pending_Get(viewModel.EntryPoint, false);
+            return Pending_Get(viewModel.EntryPoint);
         }
-
-        var questionnaire = await questionnaireService.RecordNotificationConsentAsync(
-            viewModel.CanContactByEmailAboutFutureSchemes is YesOrNo.Yes,
-            viewModel.EmailAddress
-        );
-
+        var questionnaire = questionnaireService.GetQuestionnaire();

```

Comment:
⚠️ You need to additionally save that they have ticked the confirmation checkbox, then remember that if they revisit the page (via back, etc.)